sympy/functions/elementary/trigonometric.py

```diff
@@ -855,11 +874,27 @@ def inverse(self, argindex=1):
 
     @classmethod
     def eval(cls, arg):
+        from sympy.calculus.util import Limits
         if arg.is_Number:
             if arg is S.NaN:
                 return S.NaN
             elif arg is S.Zero:
                 return S.Zero
+            elif arg is S.Infinity or arg is S.NegativeInfinity:
+                return Limits(S.NegativeInfinity, S.Infinity)
+
+        if isinstance(arg, Limits):
+            min, max = arg.min, arg.max
+            if min is not S.NegativeInfinity:
+                min = min - floor(min/S.Pi)*S.Pi
+
+            if max is not S.Infinity:
+                max = max - floor(min/S.Pi)*S.Pi
```

Comment:
I think that ``floor(min/S.Pi)`` should have the same value as above. Perhaps one should first set
`` d = floor(min/S.Pi)*S.Pi``
and then ``min -= d``, ``max -= d``.